cmd/devices-api/api.go

```diff
@@ -194,12 +194,17 @@ func startWebAPI(logger zerolog.Logger, settings *config.Settings, pdb db.Store,
 	v1Auth.Post("/user/devices", userDeviceController.RegisterDeviceForUser)
 
 	v1Auth.Get("/integrations", userDeviceController.GetIntegrations)
-	// autopi specific
-	v1Auth.Get("/autopi/unit/:unitID", userDeviceController.GetAutoPiUnitInfo)
-	v1Auth.Get("/autopi/unit/:unitID/is-online", userDeviceController.GetIsAutoPiOnline)
+
+	// Autopi specific routes.
+	apOwnerMw := owner.AutoPi(pdb, usersClient, &logger)
+	apOwner := v1Auth.Group("/autopi/unit/:unitID", apOwnerMw)
+
+	apOwner.Get("/", userDeviceController.GetAutoPiUnitInfo)
+	apOwner.Get("/is-online", userDeviceController.GetIsAutoPiOnline)
+	apOwner.Post("/update", userDeviceController.StartAutoPiUpdateTask)
```

Comment:
I think before adding the middleware here we should check whether the bodies of the handlers are undoing our hard work. And also to use our logger, etc.